admin/testbrew.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Script executed by Buildbot on the Mac mini build slave, this instantiates the
+OS X Virtual Machine to a snapshot with a clean homebrew install, then executes
+via SSH the brew update, install and test commands to verify a successful
+installation of a homebrew recipe.
+
+The host machine must have:
+        * VMWare Fusion installed,
+        * A VMWare OS X VM available at a particular location
+
+The guest virtual machine must have:
+        * A snapshot with a particular name, homebrew-clean
+        * Homebrew installed and available
+"""
+
+from _preamble import TOPLEVEL, BASEPATH
+
+import os
+import sys
+import urllib2
+
+from subprocess import check_output, CalledProcessError
+
+from flocker.provision._install import run_with_fabric, Run
+
+
+YOSEMITE_VMX_PATH = (
+    '"'
+    "{home}/Documents/Virtual Machines.localized/"
+    "OS X 10.10.vmwarevm/OS X 10.10.vmx"
+    '"'
+).format(home=os.getenv("HOME"))
+
+YOSEMITE_SNAPSHOT = "homebrew-clean"
+
+VM_USERNAME = "ClusterHQVM"
+
+VM_HOST = "10.0.126.88"
+
+
+def main():
+    try:
+        BASEPATH, TOPLEVEL
+        if len(sys.argv) < 2:
+            raise Exception("URL of homebrew recipe not specified")
+        recipe_url = sys.argv[1]
+        # Open the recipe URL just to validate and verify that it exists.
+        # We do not need to read its content.
+        urllib2.urlopen(recipe_url)
+        check_output([
+            "vmrun", "revertToSnapshot", YOSEMITE_VMX_PATH, YOSEMITE_SNAPSHOT,
```

Comment:
Why not use `vmfusion` library as per https://github.com/ClusterHQ/flocker/pull/1172/files? In the meeting IIRC it was because of a desire to have no dependencies, but that is not an issue now.